FlowCrypt/Functionality/DataManager/Encrypted Storage/EncryptedStorage.swift

```diff
@@ -181,34 +187,43 @@ extension EncryptedStorage {
     }
 
     func updateKeys(with primaryFingerprint: String, passphrase: String?) {
-        let keys = keysInfo()
-            .filter { $0.primaryFingerprint == primaryFingerprint }
+        let keys = storage.objects(KeyInfoRealmObject.self).where {
+            $0.primaryFingerprint == primaryFingerprint
+        }
 
         try! storage.write {
             keys.map { $0.passphrase = passphrase }
         }
     }
 
-    func keysInfo() -> [KeyInfoRealmObject] {
-        let result = storage.objects(KeyInfoRealmObject.self)
-        return Array(result)
+    func findKeyInfo(by email: String) -> [KeyInfo] {
+        return storage.objects(KeyInfoRealmObject.self).where({
+            $0.account == email
+        }).map(KeyInfo.init)
+    }
+
+    func findPrivateKey(by email: String) -> [String] {
+        return storage.objects(KeyInfoRealmObject.self).where({
+            $0.account == email
+        }).map(\.private)
     }
 
     func publicKey() -> String? {
         storage.objects(KeyInfoRealmObject.self)
-            .map(\.public)
-            .first
+            .first.map(\.public)
     }
```

Comment:
This method is plain wrong, it picks public key regardless of which account it belongs to.

Reply:
I'd remove it. Wherever this is used, use `getKeypairs(for email: String).map(\.public).first`